docs/arduino-deviations.rst

```diff
@@ -0,0 +1,313 @@
+Deviations from Arduino Language Reference
+==========================================
```

Comment:
Section to be reviewed once ArduinoCore-API integration.

Reply:
Section has been taken from Wiki and will be adapted incrementally